tools/serverpod_cli/lib/src/create/create.dart

```diff
@@ -94,363 +124,423 @@ Future<void> performCreate(
+      style: const TextLog(type: TextLogType.init),
+    );
+  } else if (template == ServerpodTemplateType.module) {
+    log.info(
+      '📦 Creating Serverpod module "$name"',
+      style: const TextLog(type: TextLogType.init),
+    );
+  }
 
-  log.debug(
-    'Creating directory: ${projectDir.path}',
-    style: const TextLogStyle(
-      type: AbstractStyleType.bullet,
-    ),
-  );
-  projectDir.createSync();
+  bool success = await log.progress(
+      'Setting up project structure',
+      () => Future(() {
+            _createProjectDirectories(template, serverpodDirs);
+            return true;
+          }),
+      newParagraph: true);
 
-  var serverDir = Directory(p.join(projectDir.path, '${name}_server'));
-  log.debug(
-    'Creating directory: ${serverDir.path}',
-    style: const TextLogStyle(
-      type: AbstractStyleType.bullet,
-    ),
-  );
-  serverDir.createSync();
+  if (template == ServerpodTemplateType.server) {
+    success &= await log.progress(
+        'Bootstrapping project files',
+        () => Future(() {
+              _copyServerTemplates(
+                serverpodDirs,
+                name: name,
+                awsName: awsName,
+                randomAwsId: randomAwsId,
+                dbPassword: dbPassword,
+                dbProductionPassword: dbProductionPassword,
+                dbStagingPassword: dbStagingPassword,
+              );
+              return true;
+            }));
 
-  var clientDir = Directory(p.join(projectDir.path, '${name}_client'));
-  log.debug(
-    'Creating directory: ${clientDir.path}',
-    style: const TextLogStyle(
-      type: AbstractStyleType.bullet,
-    ),
-  );
+    success &= await log.progress('Preparing your server',
+        () => CommandLineTools.dartPubGet(serverpodDirs.serverDir));
+    success &= await log.progress('Preparing your client',
+        () => CommandLineTools.dartPubGet(serverpodDirs.clientDir));
+    success &= await log.progress('Preparing your Flutter app',
+        () => CommandLineTools.flutterCreate(serverpodDirs.flutterDir));
+  } else if (template == ServerpodTemplateType.module) {
+    success &= await log.progress(
+        'Bootstrapping module files',
+        () => Future(() {
+              _copyModuleTemplates(serverpodDirs, name: name);
+              return true;
+            }));
+  }
 
-  if (template == 'server') {
-    var flutterDir = Directory(p.join(projectDir.path, '${name}_flutter'));
-    log.debug(
-      'Creating directory: ${flutterDir.path}',
-      style: const TextLogStyle(
-        type: AbstractStyleType.bullet,
-      ),
-    );
-    flutterDir.createSync();
+  if (dockerConfigured && template != ServerpodTemplateType.module) {
+    if (Platform.isWindows) {
+      success &= await CommandLineTools.cleanupForWindows(
+        serverpodDirs.projectDir,
+        name,
+      );
+    } else {
+      success &= await log.progress('Preparing your database in Docker',
```

Comment:
"Downloading and configuring Docker image." (which may better explain why this can take some time)